hefquin-engine/src/main/java/se/liu/ida/hefquin/engine/federation/access/impl/FederationAccessManagerBase1.java

```diff
@@ -161,28 +162,28 @@ protected Function<TPFResponse, CardinalityResponse> getFctToObtainCardinalityRe
 	protected static class FunctionToObtainCardinalityResponseFromSolMapsResponse implements Function<SolMapsResponse, CardinalityResponse>
 	{
 		public CardinalityResponse apply( final SolMapsResponse smResp ) {
-			final int cardinality = extractCardinality(smResp);
-			if ( cardinality < 0 )
+			try {
+				final Integer cardinality = extractCardinality( smResp );
+				return new CardinalityResponseImpl( smResp, smResp.getRequest(), cardinality );
+			} catch ( UnsupportedOperationDueToRetrievalError e ) {
```

Comment:
The try-catch block should only wrap around the function call that may produce the exception.
```suggestion
			final Integer cardinality;
			try {
				cardinality = extractCardinality( smResp );
			}
			catch ( final UnsupportedOperationDueToRetrievalError e ) {
```